crm_claim_rma/crm_claim_rma.py

```diff
@@ -0,0 +1,608 @@
+    @api.model
+    def auto_set_warranty(self):
+        """ Set warranty automatically
+        if the user has not himself pressed on 'Calculate warranty state'
+        button, it sets warranty for him"""
+        for line in self:
+            if not line.warning:
+                line.set_warranty()
+        return True
+
+    @api.model
+    def get_destination_location(self, product_id, warehouse_id):
+        """Compute and return the destination location ID to take
+        for a return. Always take 'Supplier' one when return type different
+        from company."""
+        wh_obj = self.env['stock.warehouse']
+        wh = wh_obj.browse(warehouse_id)
+        location_dest_id = wh.lot_stock_id.id
+        return location_dest_id
+
+    # Method to calculate warranty return address
+    @api.one
+    @api.model
+    def set_warranty_return_address(self):
+        """Return the partner to be used as return destination and
+        the destination stock location of the line in case of Return.
+
+        We can have various case here:
+            - company or other: return to company partner or
+              crm_return_address_id if specified
+            - supplier: return to the supplier address
+
+        """
+        return_address = None
+        psi_obj = self.env['product.supplierinfo']
+        domain = [('name', '=', self.product_id.seller_id.id),
+                  ('product_tmpl_id', '=',
+                   self.product_id.product_tmpl_id.id)]
+        seller = psi_obj.search(domain)
+        if seller:
+            return_address_id = seller.warranty_return_address.id
+            return_type = seller.warranty_return_partner
+        else:
+            # when no supplier is configured, returns to the company
+            company = self.claim_id.company_id
+            return_address = (company.crm_return_address_id or
+                              company.partner_id)
+            return_address_id = return_address.id
+            return_type = 'company'
+
+        location_dest_id = self.get_destination_location(
+            self.product_id.id,
+            self.claim_id.warehouse_id.id)
+        self.write({'warranty_return_partner': return_address_id,
+                    'warranty_type': return_type,
+                    'location_dest_id': location_dest_id})
+
+    @api.multi
+    def set_warranty(self):
+        """ Calculate warranty limit and address """
+        for claim_line_brw in self:
+            if not (claim_line_brw.product_id and claim_line.invoice_line_id):
+                raise except_orm(
+                    _('Error !'),
+                    _('Please set product or/and invoice line.'))
+            claim_line_brw.set_warranty_return_address()
+            claim_line_brw.set_warranty_limit()
+        return True
+
+
+# TODO add the option to split the claim_line in order to manage the same
+# product separately
+class crm_claim(models.Model):
+    _inherit = 'crm.claim'
+
+    def init(self, cr):
```

Comment:
@pedrobaeza I think your proposal good about ´ You should depend on crm_claim_code module (https://github.com/OCA/crm/tree/8.0/crm_claim_code) that do this better ´. I'll have RMA depends on that module.